### ISSUE TITLE
radio button always return first value

### ISSUE THREAD

**Issue by cyclops24** [perak/kitchen-site#220]
Hi @perak ,
I use below code for part of my form but radio button always return first radio value (```true```):

```
<div class="form-group field-enable">
                                <label for="enableOrNot">
                                    enable or not
                                </label>
                                <div class="input-div" data-required="true">
                                    <div class="radio">
                                        <label>
                                            <input type="radio" name="enableOrNot" value="true" data-type="bool" checked>
                                            enable
                                        </label>
                                    </div>
                                    <div class="radio">
                                        <label>
                                            <input type="radio" name="enableOrNot" value="false" data-type="bool">
                                            disable
                                        </label>
                                    </div>
                                    <span id="help-text" class="help-block">
                                    </span>
                                    <span id="error-text" class="help-block">
                                    </span>
                                </div>
                            </div>
```

I search a lot in code and guess that it is related to ***form_utils.js*** file in ***client/lib*** and ```validateForm``` function.
Thanks if you help me to fix it.

**Comment by perak** [comment]
@cyclops24 Yes, I believe problem is in `form_utils.js`, in this block:

```
// Convert to bool
if(dataType == "BOOL") {
	fieldValue = fieldValue ? true : false;
}
```
your field value is string `"true"` or string `"false"`, since `"false" ? true : false` evaluates to `true` (because `!!"false" == true`) function always returns **true** (not first value - if you change order of radiobuttons, you'll get `true` again).

Solution will be:
```
// Convert to bool
if(dataType == "BOOL") {
    if(fieldValue == "true") {
      fieldValue = true;
    } else {
      if(fieldValue == "false") {
        fieldValue = false;
      } else {
	fieldValue = fieldValue ? true : false;
      }
}
```

Please try that and if it works I'll add it to v0.9.57

**Comment by cyclops24** [comment]
Hi, @perak ,
First thanks for your rapid fix. Your fix work well and you can add it to v0.9.57.

**Comment by perak** [comment]
:+1:  ok, added (but not published yet).
